src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+    List<RunDto> allRuns = pinery.getSequencerRun().all();
+    // Filtering now helps with pulling data from a historic point in time
+    allProjectSamples = allProjectSamples.stream().filter(byCreatedBetween(null, end)).collect(Collectors.toList());
+    allRuns = allRuns.stream().filter(byCreatedBefore(end)).collect(Collectors.toList());
+
+    // First, go through and categorize all stocks.
+    // We do this in a separate step so that stocks can form the root of the Report Item,
+    // and libraries and sequencing can be added to them.
+    for (SampleDto sample : allProjectSamples) {
+      if (sample.getSampleType().contains("Library")) continue;
+      SampleDto aliquot = null;
+      SampleDto stock;
+      if (SAMPLE_CATEGORY_ALIQUOT.equals(getAttribute(ATTR_CATEGORY, sample))) {
+        aliquot = sample;
+        stock = getParent(aliquot, SAMPLE_CATEGORY_STOCK, allSamplesById);
+      } else {
+        stock = sample;
+      }
+      if (SAMPLE_CATEGORY_STOCK.equals(getAttribute(ATTR_CATEGORY, stock))) {
+        SampleDto identity = getParent(stock, SAMPLE_CATEGORY_IDENTITY, allSamplesById);
+        String externalName = getAttribute(ATTR_EXTERNAL_NAME, identity);
+        if (hasDnaAnalyteName(stock)) {
+          ReportItem stockItem = allDna.get(stock.getId());
+          if (stockItem == null) {
+            stockItem = new ReportItem(stock, externalName);
+          }
+          if (aliquot != null) stockItem.addAliquot(aliquot);
+          addReportItem(stockItem, dnaThisMonth, allDna);
+        } else {
+          ReportItem stockItem = allRna.get(stock.getId());
+          if (stockItem == null) {
+            stockItem = new ReportItem(stock, externalName);
+          }
+          if (aliquot != null) stockItem.addAliquot(aliquot);
+          addReportItem(stockItem, rnaThisMonth, allRna);
+        } // else don't care
+      }
+    }
+
+
+
+    // Then go through and categorize all libraries
+    for (SampleDto library : allProjectSamples) {
+      String type = library.getSampleType();
+
+      if (type != null && type.contains(LIBRARY) && !type.contains(" Seq") && project.equals(library.getProjectName())) {
+        SampleDto stock = getParent(library, SAMPLE_CATEGORY_STOCK, allSamplesById);
+        ReportItem match;
+        if (hasDnaAnalyteName(stock)) {
+
+          match = addLibToReportItem(stock, library, allDna, allSamplesById);
+          if (isCreatedBetween(library, start, end)) {
+            dnaThisMonth.put(stock.getId(), match);
+          }
+          allDna.put(stock.getId(), match);
+        } else {
+          match = addLibToReportItem(stock, library, allRna, allSamplesById);
+          if (isCreatedBetween(library, start, end)) {
+            rnaThisMonth.put(stock.getId(), match);
+          }
+          allRna.put(stock.getId(), match);
+        }
+      }
+    }
+
+    // Then go through all the runs and pull out lanes that contained this project's samples
+    for (RunDto run : allRuns) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)